packages/solid/dev/App.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2024 Design Barn Inc.
+ */
+
+import { createEffect, createSignal, onCleanup, type Component } from 'solid-js';
+import { type DotLottie, DotLottieSolid } from 'src';
+
+const animations = [
+  // eslint-disable-next-line no-secrets/no-secrets
+  'https://lottie.host/647eb023-6040-4b60-a275-e2546994dd7f/zDCfp5lhLe.json',
+  // eslint-disable-next-line no-secrets/no-secrets
+  'https://lottie.host/f315768c-a29b-41fd-b5a8-a1c1dfb36cd2/CRiiNg8fqQ.lottie',
+];
+
+const App: Component = () => {
+  const [dotLottie, setDotlottie] = createSignal<DotLottie | null>(null);
+  const [loop, setLoop] = createSignal(true);
+  const [speed, setSpeed] = createSignal(1);
+  const [srcIdx, setSrcIdx] = createSignal(0);
+  const [useFrameInterpolation, setUseFrameInterpolation] = createSignal(false);
+  const [playOnHover, setPlayOnHover] = createSignal(false);
+  const [autoResizeCanvas, setAutoResizeCanvas] = createSignal(true);
+
+  function play(): void {
+    dotLottie()?.play();
+  }
+
+  function pause(): void {
+    dotLottie()?.pause();
+  }
+
+  function stop(): void {
+    dotLottie()?.stop();
+  }
+
+  function handleLoop(): void {
+    setLoop((prev) => !prev);
+  }
+
+  function increaseSpeed(): void {
+    setSpeed((prev) => prev + 0.1);
+  }
+
+  function decreaseSpeed(): void {
+    setSpeed((prev) => prev - 0.1);
+  }
+
+  function changeSrc(): void {
+    const totalAnimations = animations.length;
+    const nextSrcIdx = (srcIdx() + 1) % totalAnimations;
+
+    setSrcIdx(nextSrcIdx);
+  }
+
+  function handleUseFrameInterpolation(): void {
+    setUseFrameInterpolation((prev) => !prev);
+  }
+
+  function handlePlayOnHover(): void {
+    setPlayOnHover((prev) => !prev);
+  }
+
+  function handleAutoResizeCanvas(): void {
+    setAutoResizeCanvas((prev) => !prev);
+  }
+
+  createEffect(() => {
+    dotLottie()?.addEventListener('play', console.log);
+    dotLottie()?.addEventListener('freeze', console.log);
+    dotLottie()?.addEventListener('unfreeze', console.log);
+    dotLottie()?.addEventListener('pause', console.log);
+    dotLottie()?.addEventListener('stop', console.log);
+  });
+
+  onCleanup(() => {
+    dotLottie()?.removeEventListener('play', console.log);
+    dotLottie()?.removeEventListener('freeze', console.log);
+    dotLottie()?.removeEventListener('unfreeze', console.log);
+    dotLottie()?.addEventListener('pause', console.log);
+    dotLottie()?.addEventListener('stop', console.log);
```

Comment:
This can be simplified with a guard clause check

Reply:
Resolved on https://github.com/LottieFiles/dotlottie-web/pull/234/commits/1c2353d815dc7b715f1837c645527d928698679a